### ISSUE TITLE
Effective Learning Rate/Step-Size

### ISSUE THREAD

**Issue by alexlopezcifuentes** [oval-group/dfw#3]
I am trying to obtain the effective Step-Size to compare DFW with respect to a hand-designed schedule for SGD as Figure 3 from the paper.

In the paper, it is said that the effective step-size is computed as `gamma_t * eta`. However, computing `gamma_t * eta` after each epoch seems not to change and remains always as the initial learning rate for the training. However, the model is updated and training is achieved with good results.

In addition, this line from your `epoch.py` code seems not to work as there is no `self.eta` defined inside `dfw.py`
https://github.com/oval-group/dfw/blob/15281110baac1a91e4f59a878f0dc84cd70e2ae4/src/epoch.py#L40

Any clue about this?

**Comment by lberrada** [comment]
Hi, in some instances (usually for a value of eta that is too high), the `gamma_t` factor computed by DFW keeps a value of about 1. In these cases, the update given by DFW is exactly the same as SGD with a constant learning rate, so it is possible to obtain good performance in these settings as well.

In this code, we assumed `optimizer.eta` to be unchanged throughout training, so it is defined once [here](https://github.com/oval-group/dfw/blob/master/src/optim/__init__.py#L36).

**Comment by alexlopezcifuentes** [comment]
Thanks for the quick response. 

I now see where eta value is given ([eta](https://github.com/oval-group/dfw/blob/15281110baac1a91e4f59a878f0dc84cd70e2ae4/src/optim/__init__.py#L36)), thanks for clarifying that.

For the other question, although eta is set to a high value (in my case the intial learning rate is set to 0.1), should we see any variation through training epochs in the value of `gamma_t`? Or it may happen that `gamma_t` is equal to 1 for lets say, 60-80 epochs?

**Comment by lberrada** [comment]
The evolution of `gamma_t` is almost always a non-increasing function of the iterations (usually stays at 1 for a number of iterations and then decreases), but how quickly it does all that is task-specific.

Something I have found useful is to monitor the unclipped value of `gamma_t` (that is `num / (denom + self.eps)` without the `clamp` just before [this line](https://github.com/oval-group/dfw/blob/master/src/optim/dfw.py#L101)). By monitoring this value, you should see a clear trend about the rate of decrease of `gamma_t`. In particular, this could enable you to predict how quickly `gamma_t` will go below 1 and further decrease.
